security/cmd/node_agent_k8s/main.go

```diff
@@ -160,11 +160,11 @@ var (
 			}
 
 			server, err := sds.NewServer(serverOptions, workloadSecretCache, gatewaySecretCache)
-			defer server.Stop()
 			if err != nil {
 				log.Errorf("failed to create sds service: %v", err)
 				return fmt.Errorf("failed to create sds service")
 			}
+			defer server.Stop()
```

Comment:
This is the right change, but in reality stop is never called I think, see below cmd.WaitSignal. The main go-routine should be haning here forever.

Reply:
Yes, the program would be haning cmd.WaitSignal, but the precondition is that NewServer does not return nil server.